### ISSUE TITLE
4.4.0 to 4.5.1 upgrade failed: rendered manifests contain a resource that already exists

### ISSUE THREAD

**Issue by AdamDomagalsky** [SumoLogic/sumologic-kubernetes-collection#3588]
**Describe the bug**
Getting an error on helm upgrade from sumologic 4.4.0 to 4.5.1

> Error: UPGRADE FAILED: rendered manifests contain a resource that already exists. Unable to continue with update: ServiceAccount "collection-sumologic-metrics-targetallocator" in namespace "sumologic" exists and cannot be imported into the current release: invalid ownership metadata; label validation error: key "app.kubernetes.io/managed-by" must equal "Helm": current value is "opentelemetry-operator"

**Logs** Logs which can be helpful in investigating the issue.
```bash
# Describe(sumologic/collection-sumologic-metrics-targetallocator)
Name:                collection-sumologic-metrics-targetallocator
Namespace:           sumologic
Labels:              app.kubernetes.io/component=opentelemetry-targetallocator
                     app.kubernetes.io/instance=sumologic.collection-sumologic-metrics
                     app.kubernetes.io/managed-by=opentelemetry-operator
                     app.kubernetes.io/name=collection-sumologic-metrics-targetallocator
                     app.kubernetes.io/part-of=opentelemetry
                     app.kubernetes.io/version=latest
                     chart=sumologic-4.4.0
                     heritage=Helm
                     release=collection
                     sumologic.com/app=otelcol
                     sumologic.com/component=metrics
                     sumologic.com/scrape=true
Annotations:         meta.helm.sh/release-name: collection
                     meta.helm.sh/release-namespace: sumologic
Image pull secrets:  <none>
Mountable secrets:   <none>
Tokens:              <none>
Events:              <none>
```

**Command used to install/upgrade Collection**
```bash
helm upgrade collection helm\charts\sumologic --kube-context cluster-name --namespace sumologic -f .\helm\values\cluster-name\sumologic\collection.yaml --install --create-namespace
```

**Configuration** Configuration used for Collection e.g. user-values.yaml for helm.
```bash
_helmctl:
  chart: sumologic
kube-prometheus-stack:
  prometheus:
    enabled: false
metadata:
  logs:
    autoscaling:
      maxReplicas: 1
      minReplicas: 1
    podDisruptionBudget:
      minAvailable: 0
  metrics:
    autoscaling:
      maxReplicas: 1
      minReplicas: 1
    podDisruptionBudget:
      minAvailable: 0
  persistence:
    enabled: false
metrics:
  collector:
    otelcol:
      enabled: true
  enabled: true
opentelemetry-operator:
  enabled: true
  instrumentationNamespaces: sumologic
otelcolInstrumentation:
  autoscaling:
    maxReplicas: 2
    minReplicas: 1
  enabled: true
  metrics:
    enabled: true
sumologic:
  accessId: xxx
  accessKey: yyy
  clusterName: cluster-name
  events:
    persistence:
      enabled: false
  logs:
    collector:
      otelcol:
        enabled: true
    events:
      enabled: false
    kubelet:
      enabled: false
    podDisruptionBudget:
      minAvailable: 1
    statefulset:
      replicaCount: 1
    traces:
      enabled: false
  setup:
    job:
      resources:
        limits:
          cpu: 1000m
          memory: 4024Mi
        requests:
          cpu: 1000m
          memory: 2000Mi
```

**To Reproduce** Steps to reproduce the behavior.
1. Install sumologic helm chart 4.4.0
2. Upgrade sumologic helm chart to 4.5.1

**Expected behavior**: Installation should happen with 0 downtime without a need to uninstall and reinstall again

**Environment (please complete the following information):**

- **Collection version** (e.g. `helm ls -n sumologic`): sumologic-4.4.0
- **Kubernetes version** (e.g. `kubectl version`): 1.28
- **Cloud provider**: AWS EKS

Would manual change of the label `app.kubernetes.io/managed-by` from `opentelemetry-operator` into `Helm` be a safe operation here?

**Comment by swiatekm-sumo** [comment]
This is caused by the combination of #3515 and #3539. We've let the operator manage this ServiceAccount, and later had to walk it back. Turns out Helm takes `app.kubernetes.io/managed-by` quite seriously. #3604 fixes this, and I'll issue a patch release once it's merged.

**Comment by swiatekm-sumo** [comment]
Fixed in version 4.5.2: https://github.com/SumoLogic/sumologic-kubernetes-collection/releases/tag/v4.5.2
